mtags/shared/src/main/scala-3/scala/scala/meta/internal/pc/ScalaPresentationCompiler.scala

```diff
@@ -0,0 +1,538 @@
+package scala.meta.internal.pc
+
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.CompletableFuture
+import java.util.Optional
+import java.io.File
+import java.io.BufferedWriter
+import java.io.OutputStreamWriter
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.net.URI
+import java.{util => ju}
+import scala.concurrent.ExecutionContext
+import scala.concurrent.ExecutionContextExecutor
+import scala.io.Codec
+import scala.collection.JavaConverters._
+import scala.language.implicitConversions
+import org.eclipse.lsp4j.Diagnostic
+import org.eclipse.lsp4j.Location
+import org.eclipse.lsp4j.CompletionItem
+import org.eclipse.lsp4j.CompletionItemKind
+import org.eclipse.lsp4j.CompletionList
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+import org.eclipse.lsp4j.Hover
+import org.eclipse.lsp4j.SignatureHelp
+import org.eclipse.lsp4j.SignatureInformation
+import org.eclipse.lsp4j.ParameterInformation
+import org.eclipse.lsp4j.MarkupContent
+import org.eclipse.lsp4j.FoldingRange
+import org.eclipse.lsp4j.DocumentSymbol
+import org.eclipse.lsp4j.DocumentOnTypeFormattingParams
+import org.eclipse.lsp4j.DocumentRangeFormattingParams
+import org.eclipse.lsp4j.TextEdit
+import org.eclipse.lsp4j.DiagnosticSeverity
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.interactive.Interactive
+import dotty.tools.dotc.interactive.Completion
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.core.Names._
+import dotty.tools.dotc.core.Types._
+import dotty.tools.dotc.core.Flags
+import dotty.tools.dotc.core.SymDenotations._
+import dotty.tools.dotc.core.NameOps._
+import dotty.tools.dotc.core.NameKinds._
+import dotty.tools.dotc.core.Flags._
+import dotty.tools.dotc.core.Annotations.AnnotInfo
+import dotty.tools.dotc.util.SourcePosition
+import dotty.tools.dotc.util.Spans
+import dotty.tools.dotc.util.ParsedComment
+import dotty.tools.dotc.util.Signatures
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.parsing.Parsers.Parser
+import dotty.tools.dotc.printing.PlainPrinter
+import dotty.tools.dotc.reporting.diagnostic.Message
+import dotty.tools.dotc.reporting.StoreReporter
+import dotty.tools.dotc.reporting.diagnostic.MessageContainer
+import dotty.tools.io.VirtualFile
+import dotty.tools.dotc.interfaces.{Diagnostic => DottyDiagnostic}
+import dotty.tools.dotc.ast.tpd
+import scala.meta.internal.pc.EmptyCancelToken
+import scala.meta.internal.pc.DefinitionResultImpl
+import scala.meta.internal.pc.CompilerAccess
+import scala.meta.pc.VirtualFileParams
+import scala.meta.pc.PresentationCompilerConfig
+import scala.meta.pc.PresentationCompiler
+import scala.meta.pc.SymbolSearch
+import scala.meta.pc.DefinitionResult
+import scala.meta.pc.OffsetParams
+
+case class ScalaPresentationCompiler(
```

Comment:
The code here is mostly copied from Dotty Language Server and will be mostly likely reworked heavily soon, so I wouldn't concentrate on this too much. Tests are a better  documentation showing what's currently working.

Reply:
ok :+1: